tap_pardot/client.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, creds):
         self.login()
 
         requests_session = requests.Session()
-        retries = Retry(total=100, backoff_factor=1, status_forcelist=[502, 503, 504])
+        retries = Retry(total=10, backoff_factor=0.5, status_forcelist=[502, 503, 504])
```

Comment:
still, a lot of retries, but I guess it is fine. I am still confused as to why we are doing backoff in two different ways here. You're using the @backoff decorator, and then this one. You could simply wrap the `_make_request` function with something similar to this (and then remove the header from all the other functions) but maybe something is not being captured by the decorator?:

```python
@backoff.on_exception(
    backoff.expo,
    (requests.exceptions.Timeout, requests.exceptions.ConnectionError, PardotException),
    max_tries=10,
)
```

Side note, why are you not using the `_make_request` for all requests, specifically the `login()` function?